R/plot_step_ahead_model_output.R

```diff
@@ -0,0 +1,900 @@
+          x_truth_col_name = x_truth_col_name)
+      } else if (facet == fill_by) {
+         plot_model <- simple_model_plot(
+           plot_model, df_point, df_ribbon, plot_truth, truth_data,
+           opacity = fill_transparency, top_layer = top_layer,
+           show_truth_legend = FALSE, interactive = TRUE,
+           fill_by = fill_by, x_col_name = x_col_name,
+           x_truth_col_name = x_truth_col_name)
+      } else {
+        plot_model <- simple_model_plot(
+          plot_model, df_point, df_ribbon, plot_truth, truth_data,
+          opacity = fill_transparency, showlegend = FALSE,
+          top_layer = top_layer, show_truth_legend = FALSE,
+          interactive = TRUE, fill_by = fill_by, x_col_name = x_col_name,
+          x_truth_col_name = x_truth_col_name)
+      }
+      # Ensemble color
+      if (!is.null(all_ens)) {
+        if (x == facet_value[1]) {
+          plot_model <- simple_model_plot(
+            plot_model, df_point_ens, df_ribbon_ens, FALSE, truth_data,
+            line_color = ens_color, opacity = fill_transparency,
+            top_layer = top_layer, interactive = TRUE, fill_by = fill_by,
+            x_col_name = x_col_name, x_truth_col_name = x_truth_col_name)
+        } else if (facet == fill_by) {
+          if (facet == "model_id" & ens_name == x) {
+            plot_model <- simple_model_plot(
+              plot_model, df_point_ens, df_ribbon_ens, TRUE, truth_data,
+              line_color = ens_color, opacity = fill_transparency,
+              top_layer = top_layer, show_truth_legend = FALSE,
+              interactive = TRUE, fill_by = fill_by, x_col_name = x_col_name,
+              x_truth_col_name = x_truth_col_name)
+          }
+        } else {
+          plot_model <- simple_model_plot(
+            plot_model, df_point_ens, df_ribbon_ens, FALSE, truth_data,
+            line_color = ens_color, opacity = fill_transparency,
+            showlegend = FALSE, top_layer = top_layer,
+            show_truth_legend = FALSE, interactive = TRUE, fill_by = fill_by,
+            x_col_name = x_col_name, x_truth_col_name = x_truth_col_name)
+        }
+      }
+      if (!is.null(facet_title)) {
+        if (grepl("top", facet_title)) {
+          y_title <- 1
+          y_anchor <- "top"
+        } else if  (grepl("bottom", facet_title)) {
+          y_title <- 0
+          y_anchor <- "bottom"
+        }
+        if (grepl("left", facet_title)) {
+          x_title <- 0
+          x_anchor <- "left"
+        } else if  (grepl("right", facet_title)) {
+          x_title <- 1
+          x_anchor <- "right"
+        }
+        plot_model <- plotly::layout(
+          plot_model,
+          annotations = list(x = x_title, y = y_title, xref = "paper",
+                             yref = "paper", xanchor = x_anchor,
+                             yanchor = y_anchor, showarrow = FALSE, text = x))
+      }
+      return(plot_model)
+    })
```

Comment:
This whole section is quite nested and hard to follow. I wonder if some of it can be modularised and pulled out into smaller more concise functions?

Also although the unnamed function called in `lapply` only has `x` as an argument, there are a lot of objects used from the function environment which also make it a bit confusing and hard to follow.

Reply:
I agree, it will take me some time to test and implement. I am happy to think about it for a future version.